cinema/views.py

```diff
@@ -56,3 +83,47 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        date_str = self.request.query_params.get("date")
+        if date_str:
+            _date = datetime.strptime(date_str, "%Y-%m-%d")
+            queryset = queryset.filter(show_time__date=_date)
```

Comment:
Do not make any modifications with date here, all required transformations are done under the hood in the Django ORM. Pass date as a string as is here